idaes/surrogate/alamopy_new.py

```diff
@@ -0,0 +1,504 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+from enum import Enum
+import os
+
+from idaes.surrogate.my_surrogate_base import Surrogate
+from idaes.core.util.config import list_of_ints, list_of_floats
+from pyomo.common.config import ConfigValue, In
+
+
+TEMPPATH = os.path.dirname(__file__)
+
+
+# The values associated with these must match tose expected in the .alm file
+class Modelers(Enum):
+    BIC = 1
+    MallowsCp = 2
+    AICc = 3
+    HQC = 4
+    MSE = 5
+    SSEP = 6
+    RIC = 7
+    MADp = 8
+
+
+class Screener(Enum):
+    none = 0
+    lasso = 1
+    SIS = 2
+
+
+supported_options = [

```

Comment:
Should trace and tracefname be included in this list? They are additional ALAMO options to create and name a trace .trc file, and should be supported.

Reply:
No - it is explicitly included separately.